src/dune_engine/action_builder.mli

```diff
@@ -107,10 +107,14 @@ val alias : Alias.t -> unit t
 
 val dep_on_alias_if_exists : Alias.t -> bool t
 
-(** Depend on an alias recursively. Return [true] if the alias is defined in at
-    least one directory, and [false] otherwise. *)
-val dep_on_alias_rec :
-  Alias.Name.t -> Context_name.t -> Source_tree.Dir.t -> bool t
+module Alias_rec (_ : functor (Outcome : Monoid.S) -> sig
```

Comment:
@snowleopard I made this generic to be on the safe side, but if you can confirm that you don't need this, I can move it to the rules.

Reply:
I weakly prefer keeping this functionality here for now [1], with a comment saying that we are considering to move it to rules at some point.

[1] Fundamentally, I can see that a build engine might provide some support for recursive aliases. I don't have any good concrete examples, but it seems like the engine could come up with some tricks to do it more efficiently under the hood.

Reply:
One thing that makes me suspicious is that we require the caller to be able to work with any monoid but in fact, we're only instantiating the functor with the `Exists` monoid (which, in particular, is commutative). I wonder if the overly strict requirements might make the caller awkward.

Reply:
(So, it seems we can simplify the API by doing some inlining.)

Reply:
I agree that the current API is weird. Inlining the monoid seems fine, but then we'd need to take a function of type `Path.Build.t -> f:(path:Path.Build.t -> bool t) -> bool t`, and it's hard to describe it in a way that makes a lot of sense. Perhaps by keeping it all in the rules, we can avoid this akwardness?

Reply:
Yeah, maybe this is an argument that this function belongs to the rules.

Here is an idea: define a type which is essentially `Exists | Missing`, with an appropriate `combine` function, and use it instead of `bool` both here and in `dep_on_alias_if_exists`.

Reply:
Did that.

Reply:
Looks better, thanks!